packages/core/src/models/uischema.ts

```diff
@@ -150,6 +150,16 @@ export interface SchemaBasedCondition extends BaseCondition, Scoped {
   failWhenUndefined?: boolean;
 }
 
+/** A condition using a validation function to determine its fulfillment. */
+export interface ValidateFunctionCondition extends BaseCondition, Scoped {
+  /**
+   * Validates whether the condition is fulfilled.
+   *
+   * @param data The data as resolved via the scope.
+   * @returns `true` if the condition is fulfilled */
+  validate: (data: unknown) => boolean;
```

Comment:
I think we should hand in much more data here, i.e. whatever we have available at call time.

For example
```suggestion
  validate: (rule: {
    scopedData: unknown;
    schema: JSONSchema,
    uischema: UISchemaElement,
    path: string; // or string[]
    // etc.
    }) => boolean;
```

Reply:
Generally, I agree with your sentiment. However, we do not have that much available without breaking the public API:
The `evaluateCondition` method has these parameters:
```ts
data: any, // This is the full data
condition: Condition,
path: string,
ajv: Ajv
```

Tracing upwards in the call history to the first publicly exposed methods using this - `evalVisibility` and `evalEnablement` - they have these parameters available (the same is true for `isVisible` and `isEnabled`):

```
uischema: UISchemaElement, // ui schema element containing the rule
data: any, // full data
path: string = undefined, // Optional instance path for nested controls
ajv: Ajv
```

Thus, without breaking the API of there we could additionally hand in:
- the UISchemaElement containing the condition
- the full data
- the optional instance path for nested controls

I omitted handing in the full data because this can be covered by using the root scope `#`. Thinking about it again, this is then only the nested root data.
Thus, I suggest handing in the three listed properties in addition to the scoped data. I'd like to avoid breaking the API. WDYT?

Reply:
Makes sense to me :+1: